client/packages/common/src/ui/forms/JsonForms/components/Label.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import {
+  LayoutProps,
+  rankWith,
+  UISchemaElement,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { withJsonFormsLayoutProps } from '@jsonforms/react';
+import { SxProps, Typography } from '@mui/material';
+import { RegexUtils } from '@common/utils';
+
+export const labelTester = rankWith(3, uiTypeIs('Label'));
+
+type LabelVariant = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
+
+type LayoutPropsExtended = LayoutProps & {
+  uischema: UISchemaElement & {
+    sx?: SxProps;
+    text?: string;
+    variant?: LabelVariant;
+  };
+};
+
+const variants: { [key in LabelVariant]: SxProps } = {
+  h1: { fontSize: '1.8em', fontWeight: 'bold', textAlign: 'center' },
+  h2: { fontSize: '1.4em', fontWeight: 'bold', textAlign: 'center' },
+  h3: {
+    fontWeight: 'bold',
+    textAlign: 'right',
+    width: '40%',
+    height: '1.5em', // This shouldn't be necessary 🤷‍♂️
```

Comment:
remove then?

Reply:
It *shouldn't* be necessary -- but it is. That's that weird quirk I showed you the other day where changing the % width affected the height for no apparent reason.